docs/Configuration.md

```diff
@@ -105,3 +110,145 @@ spec:
+Cloud APIs:
+
+```console
+$ gcloud iam service-accounts create ${GCP_SERVICE_ACCOUNT}
+$ gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+    --member "serviceAccount:${GCP_SERVICE_ACCOUNT}@${PROJECT_ID}.iam.gserviceaccount.com" \
+    --role ${ROLE} \
+    --project ${PROJECT_ID}
+```
+
+Get the name of your current `ProviderRevision` of this provider:
+
+```console
+$ REVISION=$(kubectl get providers.pkg.crossplane.io ${PROVIDER_GCP} -o jsonpath="{.status.currentRevision}")
+```
+
+Next, you'll configure IAM to use Workload Identity.
+In this step, you can choose one of the following options to configure service accounts:
+
+- [Option 1] Use a Kubernetes `ServiceAccount` managed by a provider's controller.
+- [Option 2] Use a Kubernetes `ServiceAccount` which you created and is specified to `.spec.serviceAccountName`
+  in a [`ControllerConfig`](https://doc.crds.dev/github.com/crossplane/crossplane/pkg.crossplane.io/ControllerConfig/v1alpha1@v1.6.2).
+
+##### 2.1. [Option 1] Use a controller-managed `ServiceAccount`
+
+Specify a Kubernetes `ServiceAccount` with the revision you got in the last
+step:
+
+```console
+$ KUBERNETES_SERVICE_ACCOUNT=${REVISION}
+```
+
+##### 2.1. [Option 2] Use a user-managed `ServiceAccount`
+
+Name your Kubernetes `ServiceAccount`:
+
+```console
+$ KUBERNETES_SERVICE_ACCOUNT=<YOUR_KUBERNETES_SERVICE_ACCOUNT>
+```
+
+Create a `ServiceAccount`, `ControllerConfig`, and `ClusterRoleBinding`:
+
+```console
+$ cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ${KUBERNETES_SERVICE_ACCOUNT}
+  namespace: upbound-system
+---
+apiVersion: pkg.crossplane.io/v1alpha1
+kind: ControllerConfig
+metadata:
+  name: ${CONTROLLER_CONFIG}
+spec:
+  serviceAccountName: ${KUBERNETES_SERVICE_ACCOUNT}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: crossplane:provider:${PROVIDER_GCP}:system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: crossplane:provider:${REVISION}:system
+subjects:
+- kind: ServiceAccount
+  name: ${KUBERNETES_SERVICE_ACCOUNT}
+  namespace: upbound-system
+EOF
+```
+
+#### 2.2. Allow the Kubernetes `ServiceAccount` to impersonate the GCP service account
+
+Grant `roles/iam.workloadIdentityUser` to the GCP service account:
+
+```console
+$ gcloud iam service-accounts add-iam-policy-binding \
+    ${GCP_SERVICE_ACCOUNT}@${PROJECT_ID}.iam.gserviceaccount.com \
+    --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[upbound-system/${KUBERNETES_SERVICE_ACCOUNT}]" \
+    --project ${PROJECT_ID}
+```
+
+Annotate the `ServiceAccount` with the email address of the GCP service account:
+
+```console
+$ kubectl annotate serviceaccount ${KUBERNETES_SERVICE_ACCOUNT} \
+    iam.gke.io/gcp-service-account=${GCP_SERVICE_ACCOUNT}@${PROJECT_ID}.iam.gserviceaccount.com \
+    -n upbound-system
+```
+
+### 3. Configure a `ProviderConfig`
```

Comment:
Hi @sergenyalcin,
Should we also include some example manifests under `examples/providerconfig`?

Reply:
Done!